src/sources/util/http.rs

```diff
@@ -124,6 +125,38 @@ impl HttpSourceAuth {
     }
 }
 
+fn decode(header: &Option<String>, mut body: Bytes) -> Result<Bytes, ErrorMessage> {
+    if let Some(encodings) = header {
+        for encoding in encodings.rsplit(',').map(str::trim) {
+            body = match encoding {
+                "identity" => body,
+                "gzip" => decode_read(GzDecoder::new(body.reader()), "gzip")?,
```

Comment:
Would there be any advantage to using [`async_compression`](https://docs.rs/async-compression/0.3.5/async_compression/) here? We use it in a few other places.

Reply:
I thought about `async_compression` too but found nothing about using it with warp :thinking: maybe land this and upgrade to `async_compression` as a separate task..?